### PR TITLE
ADS: Site Permissions section migration

### PR DESCRIPTION
**Pull request by nalcalag**
Task/Issue URL: https://app.asana.com/0/0/1203411647299660/f

### Description
Migrate Site permissions section to new components

### Steps to test this PR

- Install from this branch
- Go to Settings > Site Permissions
- [ ] Check everything looks as expected

### UI changes
| Before  | After |
| ------ | ----- |
Screenshot_20221202-114739_DuckDuckGo|Screenshot_20221201-142533_DuckDuckGo|
Screenshot_20221202-114756_DuckDuckGo|Screenshot_20221201-142543_DuckDuckGo|

### CODE REVIEW

**Comment by nalcalag** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2587** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2587" target="_blank">Graphite</a> 
    * **PR #2588** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2588" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2588?utm_source=stack-comment).

File: app/src/main/res/drawable/camera_24.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="25">
+  <path
+      android:pathData="M5.469,6.5C4.14,6.5 3.063,7.545 3.063,8.833V16.167C3.063,17.455 4.14,18.5 5.469,18.5H13.031C14.36,18.5 15.438,17.455 15.438,16.167V14.906C15.438,14.548 15.634,14.218 15.953,14.04C16.272,13.861 16.665,13.861 16.984,14.04L20.938,16.253V8.747L16.984,10.96C16.665,11.139 16.272,11.139 15.953,10.96C15.634,10.782 15.438,10.452 15.438,10.094V8.833C15.438,7.545 14.36,6.5 13.031,6.5H5.469ZM21.195,16.397L21.195,16.397L21.195,16.397ZM1,8.833C1,6.44 3.001,4.5 5.469,4.5H13.031C15.34,4.5 17.24,6.198 17.475,8.376L20.164,6.871C21.424,6.165 23,7.047 23,8.459V16.542C23,17.953 21.424,18.835 20.164,18.129L17.475,16.624C17.24,18.802 15.34,20.5 13.031,20.5H5.469C3.001,20.5 1,18.56 1,16.167V8.833Z"
+      android:fillColor="?attr/primaryIconColor"
+      android:fillAlpha="0.84"
```
Comment (malmstein) [inline app/src/main/res/drawable/camera_24.xml:25]: this has to be removed, the alpha is already in the color. Are you extracting the assets from Figma? You should get them from [Dub](https://dub.duckduckgo.com/duckduckgo/Icons/tree/master/Glyphs)

File: app/src/main/res/drawable/location_24.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  -->
+
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="25">
+  <path
+      android:pathData="M20.707,3.793C21.005,4.091 21.084,4.544 20.905,4.926L12.905,21.926C12.705,22.35 12.239,22.578 11.782,22.476C11.325,22.374 11,21.968 11,21.5V16.25C11,14.731 9.769,13.5 8.25,13.5H3C2.532,13.5 2.126,13.175 2.024,12.718C1.922,12.261 2.151,11.795 2.574,11.595L19.574,3.595C19.956,3.416 20.409,3.495 20.707,3.793ZM7.474,11.5H8.25C10.873,11.5 13,13.627 13,16.25V17.027L17.913,6.588L7.474,11.5Z"
+      android:fillColor="?attr/primaryIconColor"
```
Comment (malmstein) [inline app/src/main/res/drawable/location_24.xml:24]: ditto for color attribute and alpha

File: app/src/main/res/drawable/microphone_24.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="25">
+  <path
+      android:pathData="M12,2.535C9.81,2.535 8.035,4.31 8.035,6.5V11.5C8.035,13.69 9.81,15.465 12,15.465C14.19,15.465 15.965,13.69 15.965,11.5V6.5C15.965,4.31 14.19,2.535 12,2.535ZM9.965,6.5C9.965,5.376 10.876,4.465 12,4.465C13.124,4.465 14.035,5.376 14.035,6.5V11.5C14.035,12.624 13.124,13.535 12,13.535C10.876,13.535 9.965,12.624 9.965,11.5V6.5Z"
+      android:fillColor="?attr/primaryIconColor"
+      android:fillAlpha="0.84"
```
Comment (malmstein) [inline app/src/main/res/drawable/microphone_24.xml:25]: ditto for color attribute and alpha

File: app/src/main/res/drawable/microphone_blocked_24.xml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  -->
+
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+  <path
+      android:pathData="M8.035,6C8.035,3.81 9.81,2.035 12,2.035C14.19,2.035 15.965,3.81 15.965,6V9.785C15.965,10.244 15.529,10.664 15.08,10.762C14.492,10.892 14.035,10.424 14.035,9.821V6C14.035,4.876 13.124,3.965 12,3.965C10.876,3.965 9.965,4.876 9.965,6V11C9.965,11.598 10.223,12.136 10.634,12.508C11.063,12.896 11.398,13.652 11.146,14.172C10.979,14.517 10.589,14.726 10.245,14.557C8.935,13.909 8.035,12.56 8.035,11V6Z"
+      android:fillColor="?attr/primaryIconColor"
```
Comment (malmstein) [inline app/src/main/res/drawable/microphone_blocked_24.xml:24]: ditto for color attribute and alpha

File: app/src/main/res/layout/activity_system_search.xml [hunk 5]
```diff
@@ -104,7 +104,7 @@
                         app:layout_constraintBottom_toBottomOf="parent"
                         app:layout_constraintEnd_toStartOf="@+id/clearTextButton"
                         app:layout_constraintTop_toTopOf="parent"
-                        app:srcCompat="@drawable/ic_microphone"/>
+                        app:srcCompat="@drawable/microphone_24"/>
```
Comment (malmstein) [inline app/src/main/res/layout/activity_system_search.xml:107]: naming

File: app/src/main/res/layout/include_omnibar_toolbar.xml [hunk 6]
```diff
@@ -189,7 +189,7 @@
                         app:layout_constraintBottom_toBottomOf="parent"
                         app:layout_constraintEnd_toStartOf="@+id/clearTextButton"
                         app:layout_constraintTop_toTopOf="parent"
-                        app:srcCompat="@drawable/ic_microphone" />
+                        app:srcCompat="@drawable/microphone_24" />
```
Comment (malmstein) [inline app/src/main/res/layout/include_omnibar_toolbar.xml:192]: ditto

File: app/src/main/java/com/duckduckgo/app/sitepermissions/permissionsperwebsite/WebsitePermissionSettingType.kt [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+            return when (this) {
+                2 -> DENY
+                3 -> ALLOW
+                else -> ASK
+            }
+        }
+    }
+}
+
+enum class WebsitePermissionSettingType {
```
[the remark this answers is not in the document]
Comment (nalcalag) [reply app/src/main/java/com/duckduckgo/app/sitepermissions/permissionsperwebsite/WebsitePermissionSettingType.kt]: It's divided the use of `Type` or `Option` across the code. I would go for `Option` as well 👍 
I think it's a very good suggestion, thanks! The only think I wouldn't add is the asOptions() method, I prefer to put the options directly when building the dialog. We'll end up with plenty of `asOptions()` public methods across the app

File: app/src/main/java/com/duckduckgo/app/sitepermissions/permissionsperwebsite/WebsitePermissionSettingOption.kt [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    val icon: Int,
+    val title: Int,
+    val setting: WebsitePermissionSettingOption,
+) : Serializable
+
+enum class WebsitePermissionSettingOption(
+    val order: Int,
+    @StringRes val stringRes: Int,
+) {
+    ASK(1, R.string.permissionsPerWebsiteAskSetting),
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/sitepermissions/permissionsperwebsite/WebsitePermissionSettingOption.kt:34]: should this be a 0? weird to have a duplicated order

**Review by malmstein — APPROVED** [review]

File: app/src/main/java/com/duckduckgo/app/sitepermissions/permissionsperwebsite/WebsitePermissionSettingOption.kt (hunk 8 above)
Comment (nalcalag) [reply app/src/main/java/com/duckduckgo/app/sitepermissions/permissionsperwebsite/WebsitePermissionSettingOption.kt:34]: Talked about it. All fine 👍

**Pull request merged**